Author: Nicholas Bamber <nicholas@periapt.co.uk>
Bug: http://bugs.mysql.com/bug.php?id=65288
Date: 2012-06-02
Subject: child process picking up external config files
 The effect of the patch should be that the affected tests
 run inside a clean chroot, but are skipped where stray
 config files may upset the tests. These failed tests
 are usually indicated by mysqlcheck triggering a FATAL ERROR.
Index: mysql-5.5.38/mysql-test/include/debian-fail-local_my_cnf.inc
===================================================================
--- /dev/null
+++ mysql-5.5.38/mysql-test/include/debian-fail-local_my_cnf.inc
@@ -0,0 +1,5 @@
+if ($FOUND_LOCAL_MY_CNF)
+{
+  skip See Oracle Bug#65288 (Debian);
+}
+
Index: mysql-5.5.38/mysql-test/mysql-test-run.pl
===================================================================
--- mysql-5.5.38.orig/mysql-test/mysql-test-run.pl
+++ mysql-5.5.38/mysql-test/mysql-test-run.pl
@@ -2451,6 +2451,17 @@ sub environment_setup {
   }
 
   # ----------------------------------------------------
+  # found_local_my_cnf
+  # ----------------------------------------------------
+  my $found_local_my_cnf =
+    mtr_file_exists("~/.my.cnf") ||
+    mtr_file_exists("/etc/my.cnf") ||
+    mtr_file_exists("/etc/mysql/my.cnf");
+  if ($found_local_my_cnf) {
+    $ENV{'FOUND_LOCAL_MY_CNF'} = $found_local_my_cnf;
+  }
+
+  # ----------------------------------------------------
   # perror
   # ----------------------------------------------------
   my $exe_perror= mtr_exe_exists(vs_config_dirs('extra', 'perror'),
Index: mysql-5.5.38/mysql-test/t/mysqladmin.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/mysqladmin.test
+++ mysql-5.5.38/mysql-test/t/mysqladmin.test
@@ -1,4 +1,5 @@
 # Embedded server doesn't support external clients
+--source include/debian-fail-local_my_cnf.inc
 --source include/not_embedded.inc
 #
 # Test "mysqladmin ping"
Index: mysql-5.5.38/mysql-test/t/log_tables_upgrade.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/log_tables_upgrade.test
+++ mysql-5.5.38/mysql-test/t/log_tables_upgrade.test
@@ -1,3 +1,5 @@
+
+-- source include/debian-fail-local_my_cnf.inc
 --source include/not_embedded.inc
 --source include/have_csv.inc
 
Index: mysql-5.5.38/mysql-test/t/mysql_upgrade.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/mysql_upgrade.test
+++ mysql-5.5.38/mysql-test/t/mysql_upgrade.test
@@ -1,3 +1,5 @@
+
+-- source include/debian-fail-local_my_cnf.inc
 -- source include/mysql_upgrade_preparation.inc
 
 #
Index: mysql-5.5.38/mysql-test/t/mysql_upgrade_ssl.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/mysql_upgrade_ssl.test
+++ mysql-5.5.38/mysql-test/t/mysql_upgrade_ssl.test
@@ -1,5 +1,6 @@
 # mysql_upgrade tests requiring SSL support
 
+-- source include/debian-fail-local_my_cnf.inc
 -- source include/have_ssl_communication.inc
 -- source include/mysql_upgrade_preparation.inc
 
Index: mysql-5.5.38/mysql-test/t/plugin_auth.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/plugin_auth.test
+++ mysql-5.5.38/mysql-test/t/plugin_auth.test
@@ -1,3 +1,5 @@
+
+-- source include/debian-fail-local_my_cnf.inc
 --source include/have_plugin_auth.inc
 --source include/not_embedded.inc
 --source include/mysql_upgrade_preparation.inc
Index: mysql-5.5.38/mysql-test/suite/perfschema/t/pfs_upgrade.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/suite/perfschema/t/pfs_upgrade.test
+++ mysql-5.5.38/mysql-test/suite/perfschema/t/pfs_upgrade.test
@@ -3,6 +3,7 @@
 # database.
 # 
 
+-- source include/debian-fail-local_my_cnf.inc
 --source include/not_embedded.inc
 --source include/have_perfschema.inc
 --source include/have_lowercase0.inc
Index: mysql-5.5.38/mysql-test/suite/rpl/t/rpl_mysql_upgrade.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/suite/rpl/t/rpl_mysql_upgrade.test
+++ mysql-5.5.38/mysql-test/suite/rpl/t/rpl_mysql_upgrade.test
@@ -5,6 +5,7 @@
 # options are added into mysql_upgrade. These options control whether sql
 # statements are binlogged or not. 
 #############################################################################
+-- source include/debian-fail-local_my_cnf.inc
 --source include/master-slave.inc
 
 # Only run test if "mysql_upgrade" is found
Index: mysql-5.5.38/mysql-test/t/myisam.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/myisam.test
+++ mysql-5.5.38/mysql-test/t/myisam.test
@@ -1,3 +1,5 @@
+
+-- source include/debian-fail-local_my_cnf.inc
 #
 # Test bugs in the MyISAM code
 #
Index: mysql-5.5.38/mysql-test/t/myisampack.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/myisampack.test
+++ mysql-5.5.38/mysql-test/t/myisampack.test
@@ -1,3 +1,5 @@
+
+-- source include/debian-fail-local_my_cnf.inc
 -- disable_warnings
 DROP TABLE IF EXISTS t1,t2,t3;
 -- enable_warnings
Index: mysql-5.5.38/mysql-test/t/mysqldump.test
===================================================================
--- mysql-5.5.38.orig/mysql-test/t/mysqldump.test
+++ mysql-5.5.38/mysql-test/t/mysqldump.test
@@ -1,4 +1,5 @@
 # Embedded server doesn't support external clients
+-- source include/debian-fail-local_my_cnf.inc
 --source include/not_embedded.inc
 
 # Binlog is required
